File: Data Structures & Algorithms/add-two-numbers/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        head = ListNode()
        temp = head
        carry = 0
        while l1 and l2:
            temp.next = ListNode()
            temp=temp.next
            sum = l1.val + l2.val + carry
            carry = sum//10
            temp.val = sum%10
            l1 = l1.next
            l2 = l2.next
        while l1:
            temp.next = ListNode()
            temp=temp.next
            sum = carry+l1.val
            carry = sum//10
            temp.val = sum%10 
            l1 = l1.next
        while l2:
            temp.next = ListNode()
            temp=temp.next
            sum = carry+l2.val
            carry = sum//10
            temp.val = sum%10 
            l2 = l2.next
        if carry!=0:
            temp.next = ListNode()
            temp=temp.next
            temp.val = carry

        return head.next


                

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        int len = 0;
        ListNode temp  = head;
        while(temp != null){
            len++;
            temp = temp.next;
        }
        int x = len-n-1;
        temp = head;
        while(x>0){
            x--;
            temp = temp.next;
        }
       
        if(x==-1){
            return head.next;
        }
        temp.next = temp.next.next;
        return head;

    }
}
